docs/src/literate/ns_vs_diffeq.jl

```diff
@@ -0,0 +1,463 @@
+                    Kₑ[BlockIndex((v▄, p▄), (i, j))] += (divφ * ψ) * dΩ
+                    Kₑ[BlockIndex((p▄, v▄), (j, i))] += (ψ * divφ) * dΩ
+                end
+            end
+        end
+
+        # The last step in the element loop is to assemble `Kₑ` and `fe`
+        # into the global `K` and `f` with `assemble!`.
+        #+
+        assemble!(stiffness_assembler, celldofs(cell), Kₑ)
+    end
+    return K
+end;
+
+# ### Solution of the semi-discretized system via DifferentialEquations.jl
+# First we assemble the linear portions for efficiency. These matrices are
+# assumed to be constant over time.
+T = 10.0
+Δt₀ = 0.01
+Δt_save = 0.1
+
+M = create_sparsity_pattern(dh);
+M = assemble_mass_matrix(cellvalues_v, cellvalues_p, M, dh);
+
+K = create_sparsity_pattern(dh);
+K = assemble_stokes_matrix(cellvalues_v, cellvalues_p, ν, K, dh);
+
+# These are our initial conditions. We start from the zero solution, because it
+# is trivially admissible if the Dirichlet conditions are zero everywhere on the
+# Dirichlet boundary for t=0. Note that the time stepper is also doing fine if the
+# Dirichlet condition is non-zero and not too pathological.
+u₀ = zeros(ndofs(dh))
+apply!(u₀, ch);
+
+# At the time of writing this example we have no clean way to hook into the
+# nonlinear solver backend to apply the Dirichlet BCs. As a hotfix we override
+# the newton initialization. We cannot solve all emerging issues by developing
+# a customized newton algorithm here. This hack should only be seen as an
+# intermediate step towards integration with OrdinaryDiffEq.jl.
+function OrdinaryDiffEq.initialize!(nlsolver::OrdinaryDiffEq.NLSolver{<:NLNewton,true}, integrator)
+    ## This block is copy pasta from OrdinaryDiffEq
+    @unpack u,uprev,t,dt,opts = integrator
+    @unpack z,tmp,cache = nlsolver
+    @unpack weight = cache
+
+    cache.invγdt = inv(dt * nlsolver.γ)
+    cache.tstep = integrator.t + nlsolver.c * dt
+    OrdinaryDiffEq.calculate_residuals!(weight, fill!(weight, one(eltype(u))), uprev, u,
+                         opts.abstol, opts.reltol, opts.internalnorm, t);
+
+    # Before starting the nonlinear solve we have to set the time correctly.
+    # Note that ch is a global variable for now.
+    #+
+    update!(ch, cache.tstep);
+
+    # The update of u takes uprev + z or tmp + z most of the time, so we have
+    # to enforce Dirichlet BCs here. Note that these mutations may break the
+    # error estimators.
+    #+
+    apply!(uprev, ch)
+    apply!(tmp, ch)
+    apply_zero!(z, ch);
+
+    nothing
+end;
+
+# For the linear equations we can cleanly integrate with the linear solver
+# interface provided by the DifferentialEquations ecosystem. We use a direct
+# solver for simplicity, altough it comes with some issues. Implementing
+# GMRES with efficient preconditioner is left open for future work.
+mutable struct FerriteLinSolve{CH,F}
+    ch::CH
+    factorization::F
+    A
+end
+FerriteLinSolve(ch) = FerriteLinSolve(ch,lu,nothing)
```

Comment:
one white line between each function declaration would look better imho. Does the function need to have this `p::` closure? It looks not really beginner friendly, but I guess we are constrained to this since DiffEq calls this?

Reply:
For the second call we need this closure, because the function is stateful, carrying the sparse LU decomposition and the constraint handler.

Reply:
Sorry, question was misleading, so we define here something like a functor right? I'm just wondering if we can instead use a normal function, otherwise we should at least explain why this is needed and how it works with a link to https://docs.julialang.org/en/v1/manual/methods/index.html#Function-like-objects-1

Beginners will have a hard time figuring out what is happening in the 
```julia
function (p::FerriteLinSolve)()
```
part

Reply:
I tried to put in a normal function for the first one and got a very strange type error. Not sure what exactly is breaking. If you want you can take a closer look on this one, I will push my remaining fixes in a few minutes.

For this I just derived some quick and dirty example from https://diffeq.sciml.ai/stable/features/linear_nonlinear/#Implementing-Your-Own-LinSolve:-How-LinSolveFactorize-Was-Created

Reply:
We can just leave it as it is. But I would recommend to mention the link to the docs or at least mention functor somewhere in between